pkg/overlay/cache.go

```diff
@@ -75,9 +75,10 @@ type NodeCriteria struct {
 	AuditSuccessRatio  float64
 	UptimeCount        int64
 	UptimeSuccessRatio float64
-	Excluded           []storj.NodeID
+	Exclude            Exclude
```

Comment:
Any  reason just not to have `ExcludedNodes` and `ExcludedIPs` ?

Reply:
`cache.FindStorageNodesWithPreferences()` queries once for a number of new nodes, and again for reputable nodes. In order to not select duplicates of new node IPs, I pass in excluded IPs. I merged excluded node IDs and IPs to avoid adding another argument to the function, but I'm fine with separating them

Reply:
I think separating them is cleaner if it's not too much work for you.